src/jsonSchema.ts

```diff
@@ -0,0 +1,112 @@
+import { appendErrors, Resolver, transformToNestObject } from 'react-hook-form';
+//import * as Ajv from 'ajv';
+import Ajv, { ErrorObject, Options } from 'ajv';
+import type { JSONSchema4, JSONSchema6, JSONSchema7 } from 'json-schema';
+import {
+  FieldError,
+  FieldErrors,
+  ResolverError,
+  ResolverSuccess,
+} from 'react-hook-form/dist/types/form';
+
+export type JSONSchema = (JSONSchema4 | JSONSchema6 | JSONSchema7) & {
+  $async?: boolean;
+};
+
+const parseErrorSchema = <TFieldValues extends Record<string, any>>(
+  validationError: Array<ErrorObject> | null | undefined,
+  validateAllFieldCriteria: boolean,
+): FieldErrors<TFieldValues> =>
+  Array.isArray(validationError)
+    ? validationError.reduce(
+        (
+          previous: FieldErrors<TFieldValues>,
+          { dataPath, keyword, message },
+        ) => {
+          const path = dataPath.replace(/\//g, '.').replace(/^\./, '');
+          return {
+            ...previous,
+            ...(path
+              ? previous[path] && validateAllFieldCriteria
+                ? {
+                    [path]: appendErrors(
+                      path,
+                      validateAllFieldCriteria,
+                      previous as Record<string, any>,
+                      keyword,
+                      message,
+                    ),
+                  }
+                : {
+                    [path]:
+                      previous[path] ||
+                      ({
+                        type: keyword,
+                        message,
+                        ...(validateAllFieldCriteria
+                          ? {
+                              types: { [keyword]: message || true },
+                            }
+                          : {}),
+                      } as FieldError),
+                  }
+              : {}),
+          };
+        },
+        {},
+      )
+    : {};
+
+export interface JsonSchemaOptions {
+  ajvOptions?: Omit<Options, 'allErrors' | 'validateSchema' | 'transpile'>;
+}
+
+export const jsonSchemaResolver = <TFieldValues extends Record<string, any>>(
+  validationSchema: JSONSchema,
+  options: JsonSchemaOptions = {},
+): Resolver<TFieldValues> => {
+  if (!validationSchema || typeof validationSchema !== 'object') {
+    throw new Error('Invalid AJV schema or validation function');
+  }
+
+  const async =
+    options.ajvOptions?.async === true ||
+    [true, 'true'].includes((validationSchema as any).$async);
+  const ajv = new Ajv({
```

Comment:
This should leave at the user's app-level, not within this lib, this lib only invoke the API and return the values/errors.

Reply:
Perfect. Then it's simpler and less prone to errors. I'll change these things. Btw, do you have any idea as to why the build fails in the CI?

Reply:
no idea just yet :(

Reply:
I will take a closer look after :)

Reply:
Made a new commit to remove the async logic